### PR TITLE
Update lab03

### PR DESCRIPTION
**Pull request by CristiBM**
Replace all tasks that required knowledge of assembly with an equivalent that only uses C, add solutions, update Makefiles and coding style.

### CODE REVIEW

File: laborator/content/etapele-compilarii/solution/01-one-diy/c-standalone/name.c [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// SPDX-License-Identifier: BSD-3-Clause
+
+int puts(const char *);
```
Comment (teodutu) [inline laborator/content/etapele-compilarii/solution/01-one-diy/c-standalone/name.c:3]:
```suggestion
int puts(const char *str);
```

**Review by teodutu — COMMENTED** [review]
I'm OK with the changes so far. Exercise 7 is nice. Another nice example would be to run `make` first without a Makefile, then with a makefile with another name and lastly using implicit Makefile rules.

**Comment by teodutu** [comment]
@RazvanN7 please give your opinion as well.

File: laborator/content/etapele-compilarii/solution/01-one-diy/c-standalone/puts.asm
Comment (teodutu) [inline laborator/content/etapele-compilarii/solution/01-one-diy/c-standalone/puts.asm:1]: This file should go, together with its corresponding skeleton file.

File: laborator/content/etapele-compilarii/solution/01-one-diy/c-standalone/start.asm
Comment (teodutu) [inline laborator/content/etapele-compilarii/solution/01-one-diy/c-standalone/start.asm:1]: Ditto: replace this with a `.c` file.

File: laborator/content/etapele-compilarii/solution/05-lib-fix/add.c [hunk 2]
```diff
@@ -0,0 +1,6 @@
+// SPDX-License-Identifier: BSD-3-Clause
+
+int add(int a, int b)
+{
+	return a+b;
```
Comment (teodutu) [inline laborator/content/etapele-compilarii/solution/05-lib-fix/add.c:5]:
```suggestion
	return a + b;
```
Same for the skelton.

File: laborator/content/etapele-compilarii/solution/05-lib-fix/sub.c [hunk 3]
```diff
@@ -0,0 +1,6 @@
+// SPDX-License-Identifier: BSD-3-Clause
+
+int sub(int a, int b)
+{
+	return a-b;
```
Comment (teodutu) [inline laborator/content/etapele-compilarii/solution/05-lib-fix/sub.c:5]:
```suggestion
	return a - b;
```
+ skeleton

**Review by teodutu — CHANGES_REQUESTED** [review]
I've just noticed that exercises `01-one-{tut,diy}/c-standalone` are still using Assembly files. You need to replace them with `.c` code as well, like you did with `03-entry-fix/b-no-main2/`.

**Comment by CristiBM** [comment]
> I've just noticed that exercises `01-one-{tut,diy}/c-standalone` are still using Assembly files. You need to replace them with `.c` code as well, like you did with `03-entry-fix/b-no-main2/`.

This seems to be a bit problematic. The whole purpose of the sub-task is showing how you can link without libc and C runtime and I'm not sure how a _start() or puts() routine would be implemented in C without libc. For task 3 it was enough replacing _start() and I was able to keep libc linked; without it I couldn't have called exit() and it would have crashed.

**Comment by teodutu** [comment]
> > I've just noticed that exercises `01-one-{tut,diy}/c-standalone` are still using Assembly files. You need to replace them with `.c` code as well, like you did with `03-entry-fix/b-no-main2/`.
> 
> This seems to be a bit problematic. The whole purpose of the sub-task is showing how you can link without libc and C runtime and I'm not sure how a _start() or puts() routine would be implemented in C without libc. For task 3 it was enough replacing _start() and I was able to keep libc linked; without it I couldn't have called exit() and it would have crashed.

Yeah, you're right. Then I don't see any solution other than removing this exercise altogether :man_shrugging:.

**Comment by CristiBM** [comment]
> > > I've just noticed that exercises `01-one-{tut,diy}/c-standalone` are still using Assembly files. You need to replace them with `.c` code as well, like you did with `03-entry-fix/b-no-main2/`.
> > 
> > 
> > This seems to be a bit problematic. The whole purpose of the sub-task is showing how you can link without libc and C runtime and I'm not sure how a _start() or puts() routine would be implemented in C without libc. For task 3 it was enough replacing _start() and I was able to keep libc linked; without it I couldn't have called exit() and it would have crashed.
> 
> Yeah, you're right. Then I don't see any solution other than removing this exercise altogether man_shrugging.

I removed subtask 1-c.
Honestly, I would have voted for leaving it in, with the risk of introducing a bit of confusion, but if the general opinion is that it's not worth it, I'm ok with that :))

**Review by teodutu — APPROVED** [review]
I think it's OK now. We can revive task 1-c next semester if it turns out that removing it was exaggerated. For now, let's play it safe.

**Comment by teodutu** [comment]
Now squash these commits into a single one formatted like this: `lab/compile: Replace Assembly exercises and add solutions` so it's easier to merge.

**Pull request merged**